lib/Fmt_ast.ml

```diff
@@ -3442,11 +3442,14 @@ and fmt_functor_arg c {loc; txt= arg} =
   match arg with
   | Sugar.Unit -> Cmts.fmt c loc (str "()")
   | Sugar.Named (name, mt) ->
+      let mty = fmt_module_type c mt in
       Cmts.fmt c loc
         (wrap "(" ")"
            (hovbox 0
-              ( hovbox 0 (fmt_str_loc_opt c name $ fmt "@ : ")
-              $ compose_module (fmt_module_type c mt) ~f:Fn.id ) ) )
+              ( hovbox 0 (fmt_str_loc_opt c name $ fmt "@ :")
+              $ fmt_or (Option.is_some mty.pro) " " "@;<1 2>"
+              $ compose_module mty ~f:(fun k ->
+                    hvbox_if (Option.is_none mty.pro) 0 k ) ) ) )
```

Comment:
I think this would be easier to read if `mty.pro` was matched only once:
```ocaml
    match mty.pro with
    | Some _ -> fmt " " $ compose_module mty ~f:(fun k -> k)
    | None -> fmt "@;<1 2>" $ compose_module mty ~f:(fun k -> hvbox 0 k)
```